### ISSUE TITLE
Search results are limited to 24.

### ISSUE THREAD

**Issue by skyrim663** [jellyfin/jellyfin-web#1752]
**Describe the bug**
When I try to search for a keyword, the number of results is always 24.
Can this limit be manually modified?
At least I can change the upper limit number that suits me.
This will make the search function more convenient. 
For too many media files, limiting the search to 24 may not be good news. 

Can this limit be manually modified?

**System (please complete the following information):**
 - OS:Ubuntu 18.04
 - Virtualization: KVM
 - Clients: Browser
 - Browser: Chrome 
 - Jellyfin Version: 10.6.0
 - Playback:
 - Installed Plugins: none
 - Reverse Proxy:none
 - Base URL: none
 - Networking: [e.g. Host, Bridge/NAT]
 - Storage: [e.g. local, NFS, cloud]

**To Reproduce**

1. search for the keyword
2. Only 24 search results

**Expected behavior**

**Logs**
no error

**Comment by AlfHou** [comment]
Thanks. Where is the API endpoint on the server? I think it would make sense to have "infinite scroll" i.e. query the backend for more result every 24th item

**Comment by MrTimscampi** [comment]
> Thanks. Where is the API endpoint on the server? I think it would make sense to implement "infinite scroll" i.e. query the backend for more result every 24th item

We need a rewrite of `emby-itemscontainer.js` to have infinite scrolling. It currently doesn't support it. If you'd be up for doing that, we're all for it, as it's bring proper infinite scrolling to all the library pages.

**Comment by hawken93** [comment]
IMO there needs to be some sort of pagination because 24 is probably a good number for what you can typically display on a page, but continuously fetching more results as the user scrolls or something would be better

**Comment by SteveDinn** [comment]
This bug is critical! There are now 25 James Bond movies :)

**Comment by StraightCurlyCurves** [comment]
> This bug is critical! There are now 25 James Bond movies :)

This is the reason why I landed here! (26 with Never say Never Again) xD

**Comment by happyguy1** [comment]
ditto on the infinite search - or at very least a larger number - for instance 1024

...there are hundreds of movies with the word (let alone string) "big" in the name

**Comment by SteveDinn** [comment]
The same goes for browsing movies by their first letter: I have tons of movies that start with 'T', but I can only see the first 50 or so.

**Comment by happyguy1** [comment]
@SteveDinn - not sure what your UI experience is, but for instance, i have 330 movies that start with S, and the first 100 are displayed on the first page, with the rest displayed on successive pages
 
![image](https://user-images.githubusercontent.com/4994157/189547581-505c93d1-849a-47de-8f00-79e3e242e885.png)

**Comment by SteveDinn** [comment]
@happyguy1 You are absolutely right. When I wrote this, I was in front of my tv and had the Android TV client in mind. Apologies.

**Comment by bradbeattie** [comment]
Note with the merging of https://github.com/jellyfin/jellyfin/pull/8842, searches trigger on tags as well, which brings this limit to light again (e.g. searching a library for `tagX` might well reveal more than 24 items).

**Comment by iverlaek** [comment]
Alternatively a refinement within of the produced results would be nice also. A kind of nested search:

using a bond movie as an example: 
Step 1: searching for 'never' will create lots of results, 
Step 2: within these results you should be allowed to search again for 'say' and the 007 bond movie

**Comment by iverlaek** [comment]
@SteveDinn
 Entering 2 arguments will be an logical or. 
I'm looking for a way to have a logical and.
Sorry for the bad / unclear explanation.

**Comment by SteveDinn** [comment]
Not sure if this should apply to this bug or would require its own issue, but I also would like to be able to order search results by the same set of criteria by which I can sort things in other places:
- Alphabetically by name (or by sort name, I guess)
- by date added
- by release date
- etc.

**Comment by thornbill** [comment]
Feature requests belong on https://features.jellyfin.org

**Comment by VigibotDev** [comment]
The best search results order is by LENGTH(Name) for huge collection.

```
git diff Emby.Server.Implementations/Data/SqliteItemRepository.cs
diff --git a/Emby.Server.Implementations/Data/SqliteItemRepository.cs b/Emby.Server.Implementations/Data/SqliteItemRepository.cs
index 7921a0854..5c984f433 100644
--- a/Emby.Server.Implementations/Data/SqliteItemRepository.cs
+++ b/Emby.Server.Implementations/Data/SqliteItemRepository.cs
@@ -2987,7 +2987,8 @@ namespace Emby.Server.Implementations.Data
                 List<(string, SortOrder)> prepend = new List<(string, SortOrder)>(4);
                 if (hasSearch)
                 {
-                    prepend.Add(("SearchScore", SortOrder.Descending));
+                    //prepend.Add(("SearchScore", SortOrder.Descending));
+                    prepend.Add(("SortNameLength", SortOrder.Ascending));
                     prepend.Add((ItemSortBy.SortName, SortOrder.Ascending));
                 }

@@ -3168,6 +3169,11 @@ namespace Emby.Server.Implementations.Data
                 return ItemSortBy.SimilarityScore;
             }

+            if (string.Equals(name, "SortNameLength", StringComparison.OrdinalIgnoreCase))
+            {
+                return "LENGTH(Name)";
+            }
+
             // Unknown SortBy, just sort by the SortName.
             return ItemSortBy.SortName;
         }
```

**Comment by VigibotDev** [comment]
Since 1.9.x search become very slow (too many useless results) and this sort patch don't work.
